src/Microsoft.DotNet.Build.Tasks/GenerateCurrentVersion.cs

```diff
@@ -93,15 +94,26 @@ public override bool Execute()
             return true;
         }
 
-        public string GetCurrentRevisionForBuildName(string buildId)
+        public bool SetVersionAndRevisionFromBuildId(string buildId)
         {
-            Regex regex = new Regex(@"\.(\d+)$");
+            Regex regex = new Regex(@"(\d{8})\.(\d+)$");
+            string dateFormat = "yyyyMMdd";
             Match match = regex.Match(buildId);
-            if (match.Success && match.Groups.Count > 1)
+            if (match.Success && match.Groups.Count > 2)
             {
-                return match.Groups[1].Value;
+                DateTime buildIdDate;
+                if (!DateTime.TryParseExact(match.Groups[1].Value, dateFormat, enUS, DateTimeStyles.AssumeLocal, out buildIdDate))
```

Comment:
I tried AssumeUniversal but for some reason the comparison still required me to convert it to Universal again, so I left AssumeLocal here and if you see [this line](https://github.com/dotnet/buildtools/pull/644/files#diff-67ed61292b05c4dc9a3ee429a0e1413aR110) I am transforming it back to UTC.